packages/services/f-analytics/src/store/analytics.module.js

```diff
@@ -35,30 +43,64 @@ export default {
             requestId: '',
             orientation: '',
             display: ''
-        }
+        },
+        events: []
     }),
 
     actions: {
-        updatePlatformData: ({ commit }, platformData) => {
-            commit('pushPlatformData', platformData);
+        [PUSH_PLATFORM_DATA]: ({ commit, state }, platformData) => {
+            if (platformData) {
+                commit(PUSH_PLATFORM_DATA, platformData);
+            }
+
+            if (isDataLayerPresent()) {
+                window.dataLayer.push({ platformData: { ...state.platformData } });
+            }
+        },
+
+        [PUSH_USER_DATA]: userData => {
+            if (userData && isDataLayerPresent()) {
+                window.dataLayer.push({ userData: { ...userData } });
+            }
         },
-        updateUserData: ({ commit }, userData) => {
-            commit('pushUserData', userData);
+
+        [PUSH_PAGE_DATA]: ({ commit, state }, pageData) => {
+            if (pageData) {
+                commit(PUSH_PAGE_DATA, pageData);
+            }
+
+            if (isDataLayerPresent()) {
+                window.dataLayer.push({ pageData: { ...state.pageData } });
+            }
         },
-        updatePageData: ({ commit }, pageData) => {
-            commit('pushPageData', pageData);
+
+        [PUSH_EVENT]: ({ commit, state }, event) => {
```

Comment:
The same as for platform data here. I would split this into two actions: one for updating eventList and one for clearing it and would handle dataLayer push in the methods